Mathlib/CategoryTheory/Groupoid.lean

```diff
@@ -125,13 +139,23 @@ def Groupoid.ofHomUnique (all_unique : ∀ {X Y : C}, Unique (X ⟶ Y)) : Groupo
 
 end
 
+lemma isGroupoid_of_reflects_iso {C D : Type*} [Category C] [Category D]
+    (F : C ⥤ D) [F.ReflectsIsomorphisms] [IsGroupoid D] :
+    IsGroupoid C where
+  all_isIso _ := isIso_of_reflects_iso _ F
+
 instance InducedCategory.groupoid {C : Type u} (D : Type u₂) [Groupoid.{v} D] (F : C → D) :
```

Comment:
This one could probably be generalized to the case of a functor `F : C ⥤ D` that is equipped with a `FullyFaithful` structure, and then this instance would be obtained using `fullyFaithfulInducedFunctor`.